addons/stock/models/stock_move.py

```diff
@@ -174,9 +174,69 @@ def _default_group_id(self):
     reservation_date = fields.Date('Date to Reserve', compute='_compute_reservation_date', store=True,
         help="This is a technical field for calculating when a move should be reserved")
-    product_packaging_id = fields.Many2one('product.packaging', 'Packaging', domain="[('product_id', '=', product_id)]", check_company=True)
+    product_packaging_id = fields.Many2one(
+        'product.packaging', string='Packaging',
+        compute='_compute_product_packaging', store=True, readonly=False, precompute=True,
+        domain="[('product_id', '=', product_id)]",
+        check_company=True)
+    product_packaging_qty = fields.Float(
+        'Packaging Quantity',
+        compute='_compute_product_packaging', store=True, readonly=False, precompute=True)
     from_immediate_transfer = fields.Boolean(related="picking_id.immediate_transfer")
 
+    @api.depends('product_id', 'product_uom_qty', 'product_uom')
+    def _compute_product_packaging(self):
+        for line in self:
+            if not line.product_id or not line.product_uom_qty or not line.product_uom:
+                line.product_packaging_id = False
+            else:
+                # remove packaging if not match the product
+                if line.product_packaging_id.product_id != line.product_id:
+                    line.product_packaging_id = False
+                # remove packaging if product_uom_qty is not a multiple of packaging_uom_qty
+                if line.product_packaging_id:
+                    packaging_uom = line.product_packaging_id.product_uom_id
+                    packaging_uom_qty = line.product_uom._compute_quantity(line.product_uom_qty, packaging_uom)
+                    product_packaging_qty = packaging_uom_qty / line.product_packaging_id.qty
+                    product_packaging_qty_integer = float_round(product_packaging_qty, precision_rounding=1.0)
+                    if not float_compare(product_packaging_qty,
+                                         product_packaging_qty_integer,
+                                         precision_rounding=line.product_uom.rounding) == 0:
+                        line.product_packaging_id = False
+                # if now no packaging, try find a suitable one
+                if not line.product_packaging_id:
+                    line.product_packaging_id = line.product_id.packaging_ids._find_suitable_product_packaging(line.product_uom_qty, line.product_uom)
+
+                # compute product_packaging_qty
+                if line.product_packaging_id:
+                    packaging_uom = line.product_packaging_id.product_uom_id
+                    packaging_uom_qty = line.product_uom._compute_quantity(line.product_uom_qty, packaging_uom)
+                    line.product_packaging_qty = packaging_uom_qty / line.product_packaging_id.qty
+                else:
+                    line.product_packaging_qty = 0
```

Comment:
Again the same duplicated code. I would also not use this function always. For example I would only trigger it for planned transfer without link to SO/PO/MO